From: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Date: Mon, 14 Jan 2013 12:27:28 +0100
Subject: hpsa: fix wrong return value from hpsa_find_cfgtables

References: FATE#313833
Patch-Mainline: n/a
Make the relevant code match what's in kernel.org codebase

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/hpsa.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 83c06f6d..fb5cf05 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -4185,7 +4185,7 @@ static int __devinit hpsa_find_cfgtables(struct ctlr_info *h)
 		return -ENOMEM;
 	rc = write_driver_ver_to_cfgtable(h->cfgtable);
 	if (rc)
-		return -ENOMEM;
+		return rc;
 	/* Find performant mode table. */
 	trans_offset = readl(&h->cfgtable->TransMethodOffset);
 	h->transtable = remap_pci_mem(pci_resource_start(h->pdev,
-- 
1.7.4.2

